backend/netstorage/netstorage.go

```diff
@@ -0,0 +1,1369 @@
+		err := f.initFs(ctx, dir)
+		if err != nil {
+			if err == fs.ErrorObjectNotFound {
+				fs.Debugf(nil, "List on dir not found: %s", f.url(dir))
+				return nil, fs.ErrorDirNotFound
+			}
+			return nil, err
+		}
+	}
+
+	URL := f.url(dir)
+	if dir != "" && !strings.HasSuffix(dir, "/") {
+		dir += "/"
+	}
+	files, err := f.netStorageDirRequest(ctx, URL)
+	if err != nil {
+		return nil, err
+	}
+	for _, item := range files {
+		name := dir + f.getFileName(&item)
+		switch item.Type {
+		case "dir":
+			mtime, _ := strconv.ParseInt(item.Mtime, 10, 64)
+			when := time.Unix(mtime, 0)
+			size, _ := strconv.ParseInt(item.Bytes, 10, 64)
+			fileCount, _ := strconv.ParseInt(item.Files, 10, 64)
+			entry := fs.NewDir(name, when).SetSize(size).SetItems(fileCount)
+			entries = append(entries, entry)
+		case "file":
+			var err error
+			var entry fs.Object
+			entry, err = f.newObjectWithInfo(name, &item)
+			if err == fs.ErrorObjectNotFound {
+				return nil, err
+			}
+			entries = append(entries, entry)
+		case "symlink":
+			var entry fs.Object
+			// Add .rclonelink suffix to allow local backend code to convert to a symlink.
+			// In case both .rclonelink file AND symlink file exists, the first will be used.
+			entry, _ = f.newObjectWithInfo(name+".rclonelink", &item)
+			fs.Debugf(nil, "converting symlink to rclonelink: %s target: %s", entry.Remote(), item.Target)
+			entries = append(entries, entry)
+		default:
+			fs.Errorf(nil, "Unknown item type is: %s finalURL: %s"+item.Type, name)
+		}
+	}
+	return entries, nil
+}
+
+// ListR lists the objects and directories of the Fs starting
+// from dir recursively into out.
+//
+// dir should be "" to start from the root, and should not
+// have trailing slashes.
+//
+// This should return ErrDirNotFound if the directory isn't
+// found.
+//
+// It should call callback for each tranche of entries read.
+// These need not be returned in any particular order.  If
+// callback returns an error then the listing will stop
+// immediately.
+//
+// Don't implement this unless you have a more efficient way
+// of listing recursively that doing a directory traversal.
+func (f *Fs) ListR(ctx context.Context, dir string, callback fs.ListRCallback) (err error) {
+	if f.filetype == "" {
+		// This happens in two scenarios.
+		// 1. NewFs is done on a non-existent object, then later rclone attempts to List/ListR this NewFs.
+		// 2. List/ListR is called from the context of test_all and not the regular rclone binary.
+		err := f.initFs(ctx, dir)
+		if err != nil {
+			if err == fs.ErrorObjectNotFound {
+				fs.Debugf(nil, "ListR on dir not found: %s", f.url(dir))
+				return fs.ErrorDirNotFound
+			}
+			return err
+		}
+	}
+
+	if !strings.HasSuffix(dir, "/") && dir != "" {
+		dir += "/"
+	}
+	URL := f.url(dir)
+	u, err := url.Parse(URL)
+	if err != nil {
+		fs.Errorf(nil, "ListR: couldn't url.Parse %s: %v", URL, err)
+		return fs.ErrorDirNotFound
+	}
+
+	list := walk.NewListRHelper(callback)
```

Comment:
This looks like the storage doesn't natively support ListR? If it doesn't then don't implement this method - the higher layers will do exactly this!

Reply:
That was a temporary solution, we have full native support for ListR coming in the next poll-request.